src/Common/Abstracts/Custom_List_Table.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * The Abstract Custom List Table.
+ *
+ * @since TBD
+ *
+ * @package TEC/Common/Abstracts
+ */
+
+namespace TEC\Common\Abstracts;
+
+use TEC\Common\Admin\Abstract_Custom_List_Table as Base_Abstract;
+use TEC\Common\StellarWP\SchemaModels\Contracts\SchemaModel as Model;
+use TEC\Common\Contracts\Custom_Table_Repository_Interface as Repository;
+
+/**
+ * Class Abstract_Custom_List_Table
+ *
+ * @since TBD
+ *
+ * @package TEC/Common/Abstracts
+ */
+abstract class Custom_List_Table extends Base_Abstract {
+	/**
+	 * Returns the total number of items.
+	 *
+	 * @since TBD
+	 *
+	 * @return int The total number of items.
+	 */
+	protected function get_total_items(): int {
+		$repo = $this->get_repository();
+
+		$repo->set_use_default_args( false );
+		$result = $repo->by_args( $this->get_arguments() )->found();
+		$repo->set_use_default_args( true );
+
+		return $result;
+	}
+
+	/**
+	 * Returns the repository.
+	 *
+	 * @since TBD
+	 *
+	 * @return Repository The repository.
+	 */
+	abstract protected function get_repository(): Repository;
+
+	/**
+	 * Returns the items for the current page.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $per_page The number of items to display per page.
+	 *
+	 * @return array The items for the current page.
+	 */
+	protected function get_items( int $per_page ): array {
+		$repo = $this->get_repository();
+
+		$repo->set_use_default_args( false );
+		$results = $repo->by_args( $this->get_arguments() )->page( $this->get_pagenum() )->per_page( $per_page )->all();
+		$repo->set_use_default_args( true );
+
+		return $results;
+	}
+
+	/**
+	 * Returns whether the list is completely empty.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function is_empty(): bool {
+		$repo = $this->get_repository();
+		$repo->set_use_default_args( false );
```

Comment:
Seeing this pattern a few times, it feels like there should be a better way. I think it would be ideal to see if we can eliminate this pattern.